src/imap/scan_folders.rs

```diff
@@ -0,0 +1,75 @@
+use std::time::Instant;
+
+use crate::{config::Config, context::Context, dc_tools::time};
+use anyhow::Context as _;
+
+use crate::error::Result;
+use crate::imap::Imap;
+use async_std::prelude::*;
+
+use super::{get_folder_meaning, get_folder_meaning_by_name, FolderMeaning};
+
+impl Imap {
+    pub async fn scan_folders(&mut self, context: &Context) -> Result<()> {
+        use crate::config::Config::*;
+
+        // First of all, debounce to once per minute:
+        let mut last_scan = context.last_full_folder_scan.lock().await;
+        if let Some(last_scan) = *last_scan {
+            if last_scan.elapsed().as_secs() < 60 {
```

Comment:
Thinking about it, I actually put the `last_scan.replace(Instant::now());` to the end, as there is not much point in debouncing to 2s (for the first day after installation) otherwise (because scanning might take almost 2s on its own).